lib/amazon-pricing/ec2-di-price-list.rb

```diff
@@ -0,0 +1,52 @@
+module AwsPricing
+  class Ec2DiPriceList < PriceList
+    include AwsPricing::Ec2Common
+
+    def initialize
+      super
+      InstanceType.populate_lookups
+      get_ec2_di_od_pricing
+      get_ec2_reserved_di_pricing
+    end
+
+    protected
+
+    @@OS_TYPES = [
+        ['linux', 'linux-unix'],
+        ['rhel', 'red-hat-enterprise-linux'],
+        ['sles', 'suse-linux'],
+        ['mswin', 'windows'],
+        ['mswinSQL', 'windows-with-sql-server-standard'],
+        ['mswinSQLWeb', 'windows-with-sql-server-web'],
+        ['mswinSQLEnterprise', 'windows-with-sql-server-enterprise']
+    ]
+
+    OS_INDEX = 0
+    OD_OS_INDEX = 0
+    RI_OS_INDEX = 1
+
+    DI_OD_BASE_URL = "http://a0.awsstatic.com/pricing/1/dedicated-instances/"
+    RESERVED_DI_BASE_URL = "http://a0.awsstatic.com/pricing/1/ec2/ri-v2/"
+    RESERVED_DI_PREV_GEN_BASE_URL = "http://a0.awsstatic.com/pricing/1/ec2/previous-generation/ri-v2/"
```

Comment:
so far the pattern is to keep these base urls in aws-price-list. are you sure you want to break that pattern?